### PR TITLE
Feat/live 3675 braze integration

### PR DESCRIPTION
**Pull request by cgrellard-ledger**
### 📝 Description

With this PR, we introduce Push Notifications in LLM.
Here are the major changes :
- Major Segment upgrade from version 1.5 to 2.9
- Integration of the Braze SDK to support Push Notifications for now and later on to handle Content Cards (which will replace the recommended section in the Wallet page)
- New Modal prompted to the user in order to ask him to enable the push notifs for Ledger Live
- New Settings section to edit notifications preferences

### ❓ Context

- **Impacted projects**: `ledger-live-mobile` 
- **Linked resource(s)**: [LIVE-3675] 

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-3675]: https://ledgerhq.atlassian.net/browse/LIVE-3675?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/analytics/AnalyticsProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+}: {
+  store: any;
+  children?: ReactNode;
+}): JSX.Element | null => {
+  const [loaded, setLoaded] = useState<boolean>(false);
+  const [segmentClient, setSegmentClient] = useState<
+    SegmentClient | undefined
+  >();
+
+  const loadSegment = async (store: any) => {
```
Comment (nparigi-ledger) [inline apps/ledger-live-mobile/src/analytics/AnalyticsProvider.tsx:20]: use callback ?

File: apps/ledger-live-mobile/src/analytics/segment.ts [hunk 2]
```diff
@@ -70,14 +72,19 @@ const extraProperties = store => {
...
+  const notificationsBlacklisted = Object.entries(notifications)
+    .filter(([key, value]) => key !== "allowed" && value === false)
+    .map(([key]) => key);
 
   return {
     appVersion,
     androidVersionCode: getAndroidVersionCode(VersionNumber.buildVersion),
     androidArchitecture: getAndroidArchitecture(VersionNumber.buildVersion),
-    environment: ANALYTICS_LOGS ? "development" : "production",
+    environment: "come",
```
Comment (nparigi-ledger) [inline apps/ledger-live-mobile/src/analytics/segment.ts:84]: forgot to revert ?

Comment (cgrellard-ledger) [reply apps/ledger-live-mobile/src/analytics/segment.ts:84]: 😆

File: apps/ledger-live-mobile/src/logic/notifications.tsx [hunk 3]
```diff
@@ -390,9 +358,10 @@ const useNotifications = () => {
       drawer: "Notif",
     });
     setPushNotificationsModalOpenCallback(false);
-    navigation.navigate(NavigatorName.Settings, {
-      screen: ScreenName.NotificationsSettings,
-    });
+    handlePushNotificationsPermission();
+    // navigation.navigate(NavigatorName.Settings, {
```
Comment (nparigi-ledger) [inline apps/ledger-live-mobile/src/logic/notifications.tsx:362]: remove comments

File: package.json [hunk 4]
```diff
@@ -118,7 +118,6 @@
     },
     "patchedDependencies": {
       "react-native-fast-crypto@2.2.0": "patches/react-native-fast-crypto@2.2.0.patch",
-      "react-native-video@5.2.0": "patches/react-native-video@5.2.0.patch",
```
Comment (nparigi-ledger) [inline package.json:121]: is this confirmed ?

Comment (cgrellard-ledger) [reply package.json:121]: Absolutely not, I forgot to put it back

File: apps/ledger-live-mobile/src/analytics/segment.ts [hunk 5]
```diff
@@ -91,45 +97,43 @@ const extraProperties = store => {
...
-  }
-
+const token = ANALYTICS_TOKEN;
+export const start = async (store: any): Promise<SegmentClient | undefined> => {
   const { user, created } = await getOrCreateUser();
   storeInstance = store;
 
-  if (ANALYTICS_LOGS) console.log("analytics:identify", user.id);
+  if (created && ANALYTICS_LOGS) {
+    console.log("analytics:identify", user.id);
```
Comment (mlegall-ledger) [inline apps/ledger-live-mobile/src/analytics/segment.ts:116]: Isn't that console.log causing a TS issue ?

File: apps/ledger-live-mobile/src/notifications/HookNotifications.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { useEffect, useState, useCallback } from "react";
+import { start } from "./braze";
+
+const HookNotifications = () => {
```
Comment (mlegall-ledger) [inline apps/ledger-live-mobile/src/notifications/HookNotifications.ts:4]: Why are we naming our hooks like that ? A pretty common convention calls for hook to be prefixed with the "use" keyword. Maybe it's something we can discuss in the team ?

File: apps/ledger-live-mobile/src/analytics/segment.ts (hunk 5 above)
Comment (cgrellard-ledger) [reply apps/ledger-live-mobile/src/analytics/segment.ts:116]: No, we've got a `/* eslint-disable no-console */` at the start of the file

File: apps/ledger-live-mobile/src/notifications/HookNotifications.ts (hunk 6 above)
Comment (cgrellard-ledger) [reply apps/ledger-live-mobile/src/notifications/HookNotifications.ts:4]: Yes totally agree, for now we don't really have a clear convention for those but we can discuss it to put one in place

**Review by gre — COMMENTED** [review]
https://ledgerhq.atlassian.net/browse/LIVE-3675?focusedCommentId=246442

**Pull request merged**